Lib/encodings/idna.py

```diff
@@ -112,6 +113,11 @@ def ToUnicode(label):
             pure_ascii = True
         except UnicodeError:
             pure_ascii = False
+    if len(label) > 300:
```

Comment:
Is this a behavior change? previously a label of any length when filled with "nothing" characters is accepted as those are stripped out by nameprep. It makes no sense to have tons of those anyways, but if anyone depends on silent stripping of hundreds+ of those this would break their application. See https://datatracker.ietf.org/doc/html/rfc3454.html#section-3.1 and https://docs.python.org/3/library/stringprep.html#stringprep.in_table_b1 which `nameprep()` calls to discard characters.

Reply:
I made up this 300 limit for lack of knowing what a worst case pre-nameprep() limit should be. nameprep also does case folding character substitution. which i naively assume _might_ expand some characters into multiple?

Reply:
Please add a comment to explain this limit. Add a reference to the issue, and maybe to the RFC.

Reply:
I've expanded upon the comment with some links. i'm not happy about the state of the RFCs around all of this... but that accurately describes the messy state of IDNA in the world. I don't see a reason not to have an input limit given this is destined to be a <= 63 character output.